66-irukandji/technical_specification.md

```diff
@@ -0,0 +1,66 @@
+# Schemapack (Irukandji)
+**Epic Type:** Implementation Epic
+
+Epic planning and implementation follow the
+[Epic Planning and Marathon SOP](https://docs.ghga-dev.de/main/sops/sop001_epic_planning.html).
+
+## Scope
+
+### Outline:
+This epic aims to make datapack navigation independent of its corresponding schemapack. Currently, since the relation class is not explicitly provided in the datapack, resolving target classes requires referring back to the schemapack. This epic introduces changes to eliminate this dependency, allowing the datapack to be self-contained for navigation.
+
+Additionally, this epic introduces configurable embedding depth in datapack denormalization. This ensures more control over how deeply related entities are embedded, enabling partial embedding instead of always including all related data at the highest level.
+
+### Included/Required:
+
+#### Example datapacks/schemapacks
+
+1. Update all the examples so that the datapack also has the rootClass, targetResources, rootResource and rootClass if it is specified in schemapack. 
+
+#### Datapack specification
+
+2. Change the datapack specification 
+   1. if there is  root resource, root resource has to exits
+   2. validating against datapack specification will rely on pydantic to ensure the root resource check. 
```

Comment:
Root resource for this check has has to exist where, schemapack?
Or is this more of a "If a root resource is required/should be specified, a concrete instance has to exist"?

Please elaborate on how validation is handled if this is optional as long as the datapack is not rooted, e.g. custom validators?

Reply:
Schemapack includes two types of validation. One verifies whether a schemapack or datapack conforms to its specifications. The plug-ins, on the other hand, validate whether data provided as a datapack aligns with a given model, which is a schemapack.

Everything discussed here pertains solely to the datapack specifications. So if a resource is given as a root resource in a datapack, that resource cannot be a random string, but it has to exist among the data.

Reply:
rephrased.